File: examples/smart_contracts/v2/javascript/AccountDelegation.js
const algosdk = require('algosdk');
const fs = require('fs');
const { getAccounts } = require('./sandbox.js');


// Create an algod client, using default sandbox parameters here
const token = "a".repeat(64);
const server = "http://localhost";
const port = 4001;
let algodclient = new algosdk.Algodv2(token, server, port);

(async () => {
    // Get receiver account info from sandbox
    const accounts = await getAccounts();
    const receiverAccount = accounts[0];
    console.log("Receiver Address: " + receiverAccount.addr);


    // Get sender account info from sandbox
    const senderAccount = accounts[0];
    console.log("Sender Address: " + senderAccount.addr);

    // Get suggested parameters
    let params = await algodclient.getTransactionParams().do();
    console.log(params);

    // Read TEAL file. See more info here: https://developer.algorand.org/docs/features/asc1/sdks/#accessing-teal-program-from-sdks
    let fs = require('fs'),
        path = require('path'),
        filePath = path.join(__dirname, 'samplearg.teal');
    let data = fs.readFileSync(filePath);

    //Compile the program against the algod
    let results = await algodclient.compile(data).do();
    console.log("Hash = " + results.hash);
    console.log("Result = " + results.result);
    let program = new Uint8Array(Buffer.from(results.result, "base64"));

    // Create the lsig account, passing arg
    let args = getUint8Int(123);
    let lsig = new algosdk.LogicSigAccount(program, args);
    console.log("lsig : " + lsig.address());

    // sign the logic signature with an account sk
    lsig.sign(senderAccount.sk);

    // Setup a transaction
    let sender = senderAccount.addr;
    let receiver = receiverAccount.addr;
    let amount = 10000;
    let closeToRemaninder = undefined;
    let note = undefined;
    let txn = algosdk.makePaymentTxnWithSuggestedParams(sender,
        receiver, amount, closeToRemaninder, note, params)

    // Create the LogicSigTransaction with contract account LogicSig
    let rawSignedTxn = algosdk.signLogicSigTransactionObject(txn, lsig.lsig);
    console.log("signedTxn: ", rawSignedTxn)

    // send raw LogicSigTransaction to network    
    let tx = (await algodclient.sendRawTransaction(rawSignedTxn.blob).do());
    console.log("Transaction : " + tx.txId);

    //Wait for confirmation
    const confirmedTxn = await algosdk.waitForConfirmation(algodclient, tx.txId, 4);

    //Get the completed Transaction
    console.log("Transaction " + tx.txId + " confirmed in round " + confirmedTxn["confirmed-round"]);


})().catch(e => {
    console.log(e.message);
    console.log(e);
});

function getUint8Int(number) {
    const buffer = Buffer.alloc(8);
    const bigIntValue = BigInt(number);
    buffer.writeBigUInt64BE(bigIntValue);
    return [Uint8Array.from(buffer)];
}

File: examples/smart_contracts/v2/python/account_delegation.py
import os
import base64
import json

from algosdk.v2client import algod
from algosdk.future.transaction import *
from algosdk import transaction
from algosdk.atomic_transaction_composer import *
from algosdk.constants import *
from algosdk.util import *

# Note this method is defined in sandbox.py
from sandbox import get_accounts

# This code is meant for learning purposes only
# It should not be used in production

# Read a file
def load_resource(res):
    dir_path = os.path.dirname(os.path.realpath(__file__))
    path = os.path.join(dir_path, res)
    with open(path, "r") as f:
        data = f.read()
    return data


try:
    # Create an algod client, using default sandbox parameters here
    algod_token = "a" * 64
    algod_address = "http://localhost:4001"
    algod_client = algod.AlgodClient(algod_token, algod_address)

    # Get receiver account info (address and private key)
    receiver_addr, receiver_sk = get_accounts()[0]

    # Load in our program
    myprogram = "samplearg.teal"
    data = load_resource(myprogram)

    # Compile the program against the algod
    response = algod_client.compile(data)
    print(f"Response Result (base64 encoded): {response['result']}")
    print(f"Response Hash: {response['hash']}")

    # Decode the program to bytes and encode the argument as bytes
    program = base64.b64decode(response["result"])
    arg1 = (123).to_bytes(8, "big")

    # Create the lsig account, passing arg
    lsig = LogicSigAccount(program, args=[arg1])

    # Recover the account that is wanting to delegate signature
    # never use mnemonics in code, for demo purposes
    delegator_addr, delegator_sk = get_accounts()[1]
    print("Address of Sender/Delegator: " + delegator_addr)

    # Sign the logic signature with an account sk
    lsig.sign(delegator_sk)

    # Get suggested parameters
    params = algod_client.suggested_params()

    # replace with any other address or amount
    receiver = receiver_addr
    amount = 10000

    # Create a transaction
    txn = PaymentTxn(delegator_addr, params, receiver, amount)

    # Create the LogicSigTransaction with contract account LogicSigAccount
    lstx = transaction.LogicSigTransaction(txn, lsig.lsig)

    # Send raw LogicSigTransaction to network
    txid = algod_client.send_transaction(lstx)
    print("Transaction ID: " + txid)

    # Wait for confirmation
    confirmed_txn = wait_for_confirmation(algod_client, txid, 4)
    print(f"Result confirmed in round: {confirmed_txn['confirmed-round']}")
    print(f"Transaction information: {json.dumps(confirmed_txn, indent=4)}")
except Exception as e:
    print(e)
